packages/@angular/cli/update/check.ts

```diff
@@ -0,0 +1,143 @@
+import * as path from 'path';
+import * as https from 'https';
+import {lt} from 'semver';
+
+
+const ONE_HOUR = 3600000;
+const ONE_DAY = ONE_HOUR * 24;
+const ONE_WEEK = ONE_DAY * 7;
+
+export interface CheckOptions {
+  includeUnstable: boolean;
+}
+
+export interface CheckResult {
+  currentVersion: string;
+  newVersion?: string;
+}
+
+export function timeToCheck(lastChecked: Date, frequency: string): boolean {
+  const freq = frequency || '1d';
+  if (freq.toLowerCase() === 'never') {
+    return false;
+  } else if (freq.toLowerCase() === 'always') {
+    return true;
+  }
+  if (!lastChecked) {
+    lastChecked = new Date(0);
+  }
+  const diff = Math.abs(lastChecked.valueOf() - new Date().valueOf());
+  const threshold = convertFrequency(freq);
+  return diff > threshold;
+}
+
+export function newVersionAvailable(
+  rootDir: string,
+  name: string,
+  options?: CheckOptions): Promise<CheckResult> {
+  return Promise.all([
+      getLocalVersion(rootDir, name),
+      getRemoteVersion(name, options)
+    ])
+    .then(([localVersion, remoteVersion]) => {
+      const newer = lt(localVersion, remoteVersion);
+      const result: CheckResult = {
+        currentVersion: localVersion
+      };
+      if (newer) {
+        result.newVersion = remoteVersion;
+      }
+      return result;
+    });
+}
+
+function encodePackageName(name: string): string {
+  let encoded = encodeURIComponent(name);
+  if (name.startsWith('@')) {
+    encoded = `@${encoded.substr(3)}`;
+  }
+  return encoded;
+}
+
+function getRemoteVersion(name: string, options?: CheckOptions): Promise<string> {
+  const url = `https://registry.npmjs.org/${encodePackageName(name)}`;
+  return httpsGet(url)
+    .then(resp => {
+      const version = options.includeUnstable ?
+        resp['dist-tags'].next :
+        resp['dist-tags'].latest;
+      return version;
+    });
+}
+
+function getLocalVersion(rootDir: string, name: string): Promise<string> {
+  const packagePath = path.resolve(rootDir, 'node_modules', name, 'package.json');
+  return new Promise((resolve, reject) => {
+    try {
```

Comment:
Looks like there's no asynchronous workflow in this function. You could just return `Promise.resolve(packageInfo.version)`.